### ISSUE TITLE
Add native Github annotations output

### ISSUE THREAD

**Issue by dmohns** [sqlfluff/sqlfluff#3106]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### Description

Currently SQLFluff's `github-annotation` output is a custom JSON output tailored for this https://github.com/yuzutech/annotations-action Github Action.

Reading through the related issue and PR it's not really clear to my what is the advantage of using this additional Github Action over just writing native Github Action annotation output.

A good example is yamllint's `-f github` option. 

```sh
❯ yamllint 1.yaml -f github
::error file=1.yaml,line=3,col=3::syntax error: expected <block end>, but found '<block mapping start>' (syntax)
```

This can be integrated into Github actions extremely easy.

### Use case

Simplify Github Actions workflows be requiring one less action.

Additionally: I haven't tested it a lot, but currently it looks like it's not possible to have the CI report failure _and_ post annotations. For the annotations action to work, we need to run SQLFluff with `--nofail`.

### Dialect

all

### Are you willing to work on and submit a PR to address the issue?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by dmohns** [comment]
Initial support for this was added in https://github.com/sqlfluff/sqlfluff/pull/3107 via `github-annotation-native` output format.

One question remained open, so I'm taking it here for visibility:

Do wo we want to continue to maintain both output formats `github-annotation` and `github-annotation-native`?

**Comment by barrywhart** [comment]
@emilyhawkins-drizly: If I recall correctly, you werethe first user of the earlier annotation format. Any thoughts on whether to keep it?

Personally, I'd like to depreciate and eventually remove it, but I'm not adamant about it.

**Comment by tunetheweb** [comment]
Let's close this and track as separate issue as the issue title has been addressed.
